simpleflow/workflow.py

```diff
@@ -27,7 +32,21 @@ def submit(self, func, *args, **kwargs):
             :rtype: Future.
 
         """
-        return self._executor.submit(func, *args, **kwargs)
+        # If the func is a child workflow, call directly
+        # the executor
+        if inspect.isclass(func) and issubclass(func, Workflow):
+            return self._executor.submit(func, *args, **kwargs)
+        elif isinstance(func, task.ActivityTask):
+            return self._executor.submit(func.activity, *func.args, **func.kwargs)
+        elif isinstance(func, activity.Activity):
+            return self._executor.submit(func, *args, **kwargs)
+        elif isinstance(func, (canvas.Group, canvas.Chain)):
```

Comment:
canvas.Chain is a child of canvas.Group, `isinstance(my_chain, canvas.Group)` will return True